yt/yt/server/http_proxy/coordinator.cpp

```diff
@@ -522,6 +522,34 @@ void THostsHandler::HandleRequest(
     }
 }
 
+
+////////////////////////////////////////////////////////////////////////////////
+
+TClusterConnectionHandler::TClusterConnectionHandler(NApi::IClientPtr client)
+    : Client_(std::move(client))
+{ }
+
+void TClusterConnectionHandler::HandleRequest(
+    const NHttp::IRequestPtr& req,
+    const NHttp::IResponseWriterPtr& rsp)
+{
+    auto path = req->GetUrl().Path;
+    if (path != "/cluster_connection" && path != "/cluster_connection/") {
+        rsp->SetStatus(EStatusCode::NotFound);
+        WaitFor(rsp->Close())
+            .ThrowOnError();
+        return;
+    }
+
+    auto nodeYson = WaitFor(Client_->GetNode("//sys/@cluster_connection"))
```

Comment:
Let's add here an option to read from master cache. By default, it will be a follower.